CRM/Core/DAO.php

```diff
@@ -1322,14 +1322,16 @@ public static function &singleValueQuery(
   }
 
   /**
-   * @param $query
+   * Compose the query by merging the parameters into it.
+   *
+   * @param string $query
    * @param array $params
    * @param bool $abort
    *
    * @return string
    * @throws Exception
    */
-  public static function composeQuery($query, &$params, $abort = TRUE) {
+  public static function composeQuery($query, $params, $abort = TRUE) {
```

Comment:
This initially made me a nervous about hidden side-effects, but in reading `composeQuery`... `$params` is only referneced once (`foreach ($params...)`)... in a way which prevents one modifying values in it. So agree this appears safe.